### DISCUSSION TITLE
Adding An Option for CRANE to Perform Linear Interpolation for EEDF Rates

### DISCUSSION THREAD

**Discussion by csdechant** [shannon-lab/zapdos#86] [Q&A Plasma Chemistry / CRANE]
Just as the title suggested, would it be worth adding an option to CRANE to perform linear interpolation when using "ZapdosEEDFRate"? I know that CRANE used to uses linear interpolation and switch to spline (which is mostly likely best in most situations), but it seems that spline gives poor results when extrapolating out for large amount of data points. For some of the validation work, the comparison work had the reaction rates for a limited range of energies, and spline couldn't handle the extrapolation (it rates would eventually plummet to zero to high energy, Te ~ 14 eV) . I modified my CRANE branch to use linear interpolation ( https://github.com/csdechant/crane/tree/LinearInterpolation ), but I only added the material object and haven't added the option to the CRANE/Zapdos action. 

@keniley1 - Do you think this would be worth adding the CRANE? I also could just linear extrapolate the data in Python and feed that to CRANE using spline, but thought the option to switch between the search types could be helpful later on.

**Comment by lindsayad** [comment]
seems like a good addition to me!

**Comment by keniley1** [comment]
Yeah, I also think this would be a good addition. I've actually added a linear interpolation option already for scalar simulations (e.g. using Crane in 0D). I have noticed that splines tend to go crazy sometimes, especially at low energy (I think it's because splines are not guaranteed to preserve monotonicity?) but it never became a major issue so I haven't gotten around to working it in to the Materials system for Zapdos. 

I think this will be a relatively quick thing to add. I can just add an input parameter to the action: 

`interpolation_type = 'linear'`

If that works for you, I'll get on this asap. I can get the update pushed by this weekend.

My only concern is the discontinuous derivatives you get with linear interpolations; @lindsayad  do you think the AD system struggle with that or not?

EDIT: Created an issue for this: 
https://github.com/lcpp-org/crane/issues/70

**Comment by lindsayad** [comment]
Only if you try to evaluate the function at a kink!

**Comment by keniley1** [comment]
As usual, this took approximately 5 times longer than I suggested. @csdechant I submitted a PR to Crane to address your request. https://github.com/lcpp-org/crane/pull/71

Please try it out to make sure it works for you! I only included AD versions so you'll need to use the `use_ad = true` input parameter in the Reactions block. I figured since we're moving Zapdos to pure AD anyway it wouldn't be necessary to have a non-ad version, but if you're not using AD  yet let me know. I can add non-AD versions if necessary. 

Once that is merged I'll also submit a PR to update the version of Crane in Zapdos along with a couple tests for the new input parameter.
